### PR TITLE
WIP : Add Scala 2.13 support

### PR DESCRIPTION
**Pull request by tampler**
Few tests fail for 2.13

TODO:
1. Review and fix errors in Specs
2. Add 2.13 target in CI

### CODE REVIEW

File: src/main/scala/firrtl/transforms/PropagatePresetAnnotations.scala [hunk 1]
```diff
@@ -8,7 +8,9 @@ import firrtl.annotations._
 import firrtl.ir.{AsyncResetType, _}
 import firrtl.options.Dependency
 
-import scala.collection.mutable
+import scala.collection.mutable.{ ArrayBuffer, HashMap, HashSet }
```
Comment (ekiwi-sifive) [inline src/main/scala/firrtl/transforms/PropagatePresetAnnotations.scala:11]: Why did you change this import? It is standard in the `firrtl` code base to only import `scala.collection.mutable` and then prefix all types with `mutable.`.

File: src/main/scala/firrtl/Emitter.scala [hunk 2]
```diff
@@ -464,19 +467,6 @@ class VerilogEmitter extends SeqTransform with Emitter {
       "(2001). Use the SystemVerilog emitter instead.")
   }
   
-  /** 
-    * Store Emission option per Target
-    * Guarantee only one emission option per Target 
-    */
-  private[firrtl] class EmissionOptionMap[V <: EmissionOption](val df : V) extends collection.mutable.HashMap[ReferenceTarget, V] {
```
Comment (ekiwi-sifive) [inline src/main/scala/firrtl/Emitter.scala:471]: Why did you delete this? I suspect that this is one of the causes for the `PresetSpec` failing. `Preset` relies on `EmissionOption`s.

File: build.sbt [hunk 3]
```diff
@@ -2,55 +2,23 @@
...
-    "-Yrangepos",          // required by SemanticDB compiler plugin
-    "-Ywarn-unused-import" // required by `RemoveUnused` rule
   ),
-  javacOptions ++= javacOptionsVersion(scalaVersion.value),
   libraryDependencies ++= Seq(
     "org.scala-lang" % "scala-reflect" % scalaVersion.value,
     "org.scalatest" %% "scalatest" % "3.1.2" % "test",
     "org.scalatestplus" %% "scalacheck-1-14" % "3.1.1.1" % "test",
-    "com.github.scopt" %% "scopt" % "3.7.1",
+    "com.github.scopt" %% "scopt" % "4.0.0-RC2",
```
Comment (ekiwi-sifive) [inline build.sbt:21]: Does the stable version of `scopt` not support `2.13`?

File: build.sbt [hunk 4]
```diff
@@ -192,3 +161,9 @@ lazy val benchmark = (project in file("benchmark"))
     test in assembly := {},
     assemblyOutputPath in assembly := file("./utils/bin/firrtl-benchmark.jar")
   )
+
+// Aliases
```
Comment (ekiwi-sifive) [inline build.sbt:165]: Why is this part of a PR for upgrading to `2.13`?
Could you please remove this to make this PR easier to review?

File: project/plugins.sbt [hunk 5]
```diff
@@ -20,7 +20,7 @@ addSbtPlugin("com.simplytyped" % "sbt-antlr4" % "0.8.2")
 
 addSbtPlugin("com.github.gseitz" % "sbt-protobuf" % "0.6.5")
 
-addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.15")
+addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.18-1")
```
Comment (ekiwi-sifive) [inline project/plugins.sbt:23]: Does the old version not work with `2.13`?

File: build.sbt [hunk 6]
```diff
@@ -2,55 +2,23 @@
...
-  scalacOptions := scalacOptionsVersion(scalaVersion.value) ++ Seq(
+  crossScalaVersions := Seq("2.11.12", "2.12.11", "2.13.3"),
+  scalacOptions := Seq(
     "-deprecation",
     "-unchecked",
     "-language:reflectiveCalls",
-    "-language:existentials",
     "-language:implicitConversions",
-    "-Yrangepos",          // required by SemanticDB compiler plugin
-    "-Ywarn-unused-import" // required by `RemoveUnused` rule
```
Comment (ekiwi-sifive) [inline build.sbt:46]: Why did you remove this? Is this related to you upgrading `scalafix`?

File: src/main/scala/firrtl/Emitter.scala (hunk 2 above)
Comment (ekiwi-sifive) [reply src/main/scala/firrtl/Emitter.scala:471]: I see you moved it to compat, but it seems like you might have changed the implementation.

**Comment by sequencer** [comment]
Would you mind rebasing this PR on to master? I think it difference is a little big to review now:(

**Comment by tampler** [comment]
I made rebase, but it returned me few errors after reporting the successful rebase. 
I'll close this PR and reopen a new one to make it cleaner

Thanks for reviewing this !
